syft/generic/pointers/pointer_tensor.py

```diff
@@ -274,6 +274,9 @@ def move(self, destination: AbstractWorker, requires_grad: bool = False):
         if self.owner.id == destination.id:
             return self.get()
 
+        if self.location.id == destination.id:
```

Comment:
I think this should be handle at a lower level, maybe on the reception of a Communication Message.
Otherwise, we may need this kind of check at several places (for different methods doing doing stuff with pointers). What do you think?

Reply:
There's a similar check in the de-serialization, but I don't think it's sufficient to resolve this issue. Not sure where the right place for this is. I suppose we could add a decorator that wraps the function with this check and the one above? Not sure how else to avoid duplication.

Reply:
I don't think it should be done in serde, it's too late at this point.
I think it should be done on the worker side, when a worker receives the order to send a tensor (basically all the communication action will resume to that at the end, right?) it should check the source and destination of the communication and do things accordingly to that.
But it's just how I see it, maybe there's better?

Reply:
I think this is even before a message gets sent; it's checking whether the worker has been asked to move a tensor to itself and bypassing sending a message to itself in that case. I can't think of a better place to do this yet, but I'm guessing we'll figure it out as we rework the communication methods for `Protocols`. 🤷